tasm/semantic_index/include/SemanticIndex.h

```diff
@@ -60,8 +60,12 @@ class SemanticIndex {
 
 class SemanticIndexSQLite : public SemanticIndex {
 public:
-    SemanticIndexSQLite(std::experimental::filesystem::path dbPath = "labels.db") {
-        openDatabase(dbPath);
+    SemanticIndexSQLite(bool inMemory = false, std::experimental::filesystem::path dbPath = "labels.db") {
```

Comment:
I think this is a great starting point! The one thing that I'm not sure about is having to specify a labels path _and_ that it's not in-memory. It seems like if a labels path is specified, then it's implied that it's not an in-memory database.

I have two ideas for making this cleaner, but I'm interested if you have any others.
My first idea is to switch the dbPath argument to `std::optional<path>`. If the optional is empty, we'll assume the user wants to create an in-memory database. Otherwise, we'll get the path from the optional and use that like we currently do.

My second idea makes creating an in-memory database more explicit. We could create a subclass of `SemanticIndexSQLite`, e.g. `SemanticIndexSQLiteInMemory`. All of the functions would remain the same, except the constructor would take no arguments, and we would override `openDatabase` to open the db in-memory. With this approach, your modification to TASM's constructor could stay mostly the same, except we would construct a new semantic index of the correct type instead of passing on the `inMemory` argument.